### PR TITLE
Add Contributor Covenant Code of Conduct

### PR DESCRIPTION
**Pull request by shazow**
We had an informal discussion about this on the Discord, so consider this a starting point. :) 

If anyone has proposals for a different CoC or any modifications to this one, we'd love to hear them!

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at @sethmlarson or @shazow. All
```
Comment (shazow) [inline CODE_OF_CONDUCT.md:58]: Typically you put an email address here, but perhaps this is sufficient?

Comment (shazow) [reply CODE_OF_CONDUCT.md:58]: How wild would it be to have a maintainers@urllib3.dev email address... >.>

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+obligated to maintain confidentiality with regard to the reporter of an incident.
+Further details of specific enforcement policies may be posted separately.
+
+Project maintainers who do not follow or enforce the Code of Conduct in good
+faith may face temporary or permanent repercussions as determined by other
+members of the project's leadership.
+
+## Attribution
+
+This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4,
```
Comment (pquentin) [inline CODE_OF_CONDUCT.md:70]: Should we use version 2.0 instead?

Change log: https://github.com/ContributorCovenant/contributor_covenant/blob/release/changelog

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (pquentin) [reply CODE_OF_CONDUCT.md:58]: You both list an email address from your GitHub profile so that sounds good to me

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (shazow) [reply CODE_OF_CONDUCT.md]: Done.

**Review by pquentin — APPROVED** [review]

**Review by sethmlarson — APPROVED** [review]

**Comment by sethmlarson** [comment]
Consider us code-of-conduct-ified :tada: Thanks @shazow!

**Pull request merged**